plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+
+## Bring your own Query Client
+
+By default, `CoderProvider` manages and maintains its own `QueryClient` instance for managing all ongoing queries and mutations. This client is isolated from any other code, and especially if you are only using official Coder components, probably doesn't need to be touched.
+
+However, let's say you're using Tanstack Query for other purposes and would like all Coder requests to go through your query client instance instead. In that case, you can feed that instance into `CoderProvider`, and it will handle all requests through it instead
+
+```tsx
+<CoderProvider
+  appConfig={yourCustomAppConfig}
+  queryClient={yourCustomQueryClient}
+>
+  <YourCustomComponents />
+</CoderProvider>
+```
+
+### Warning
+
+Be sure to provide a custom query client if you put any custom components inside `CoderProvider`. Because of React Context rules, any calls to Tanstack APIs (including the `useCoderQuery` convenience wrapper) will go through the default `CoderProvider` query client, rather than your custom client.
+
+```tsx
+const customQueryClient = new QueryClient();
+
+function YourCustomComponent() {
+  const client = useQueryClient();
+  console.log(client);
+
+  return <p>Here's my content!</p>;
+}
+
+<QueryClient client={yourCustomQueryClient}>
+  {/* When mounted here, component receives customQueryClient */}
+  <YourCustomComponent />
+
+  {/*
+   * Forgot to thread customQueryClient into CoderProvider. All children will
+   * receive the default query client that CoderProvider maintains
+   */}
+  <CoderProvider>
+    {/*
+     * Exact same component definition, but the location completely changes the
+     * client that gets accessed
+     */}
+    <YourCustomComponent />
+  </CoderProvider>
+</QueryClient>;
+```
+
+## Sharing data between different queries and mutations
+
+Internally, the official components for the Coder plugin use one shared query key prefix value (`CODER_QUERY_KEY_PREFIX`) for every single query and mutation. This ensures that all Coder-based queries share a common root, and can be easily vacated from the query cache in response to certain events like the user unlinking their Coder account (basically "logging out").
+
+This same prefix is exported to users of the plugin. When using the Coder SDK with Tanstack Query, it is **strongly, strongly** recommended that you use this same query prefix. It lets custom components and Coder components share the same underlying data (reducing the total number of requests). But the plugin is also set up to monitor queries with this prefix, so it can automate cache management and has more ways of detecting expired auth tokens.
+
+## Performing queries
```

Comment:
I think we could be prescriptive and just recommend `useCoderQuery`, replacing this with a React Query example above.